.github/workflows/dispatch-yocto-build-deploy.yml

```diff
@@ -0,0 +1,124 @@
+name: "Test Yocto"
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+    branches:
+      - "main"
+      - "master"
+  pull_request_target:
+    types: [opened, synchronize]
+    branches:
+      - "main"
+      - "master"
+
+  workflow_dispatch:
+    # you may only define up to 10 `inputs` for a `workflow_dispatch` event
+    inputs:
+      device-repo:
+        description: balenaOS device repository (owner/repo)
+        required: true
+        type: string
+        default: balena-os/balena-generic
+      device-repo-ref:
+        description: balenaOS device repository tag, branch, or commit to build
+        required: false
+        type: string
+        default: master
+      meta-balena-ref:
+        description: meta-balena ref if not the currently pinned version
+        required: false
+        type: string
+      yocto-scripts-ref:
+        description: balena-yocto-scripts ref if not the currently pinned version
+        required: false
+        type: string
+      machine:
```

Comment:
Again, this is something that we should probably just extract from the device repository.

Reply:
We will call the shared workflow once per machine that we want to build, so it will be hardcoded and not exposed unless we want to allow manual execution via the device repos directly, but that's a lot of parameter duplication.